### ISSUE TITLE
[App]: Sweep run defaults to random search super early due to some dimensions of kernel are close to their bounds

### ISSUE THREAD

**Issue by UliPrantz** [wandb/wandb#8149]
### Current Behavior

I started a sweep run of a BERT model called MedBERT with the following sweep configuration:

```
method: bayes
metric:
  goal: maximize
  name: validation_auroc_score.max
parameters:
  batch_size:
    distribution: categorical
    values:
      - 16
      - 32
      - 64
      - 128
  bert_base_path:
    value: ../data/processed/trained_tensorflow_small
  bert_checkpoint:
    distribution: categorical
    values:
      - 1500000
      - 2000000
      - 2500000
      - 3000000
      - 4000000
      - 5000000
      - 6000000
      - 7000000
      - 8000000
      - 10000000
  classifier:
    value: RNN
  classifier_dropout_rate:
    distribution: uniform
    max: 0.8
    min: 0.1
  classifier_num_hidden_layers:
    distribution: int_uniform
    max: 10
    min: 1
  classifier_size_hidden_layers:
    distribution: categorical
    values:
      - 16
      - 32
      - 64
      - 128
  drop_last_batch:
    value: true
  epochs:
    distribution: normal
    mu: 15
    sigma: 5
  lr:
    distribution: uniform
    max: 0.0001
    min: 1e-07
  min_delta:
    value: 0.001
  path_fine_tuning_train_dataset:
    value: ../data/processed/sepsis/sepsis_train_feature_list.pkl
  patience:
    value: 20
  shuffle_batch:
    value: true
  train_bert:
    value: true
  use_test_set:
    value: false
run_cap: 200
```

Which results in the following log of WandB

```
Sending commands to agent ngtqkr2l: [{"run_id":"ez9wh398","program":"","type":"run","args":{"batch_size":{"value":128},"bert_base_path":{"value":"../data/processed/trained_tensorflow_small"},"bert_checkpoint":{"value":7000000},"classifier":{"value":"RNN"},"classifier_dropout_rate":{"value":0.5408377810623216},"classifier_num_hidden_layers":{"value":3},"classifier_size_hidden_layers":{"value":16},"drop_last_batch":{"value":true},"epochs":{"value":13.393495646775497},"lr":{"value":0.00006699547970432092},"min_delta":{"value":0.001},"path_fine_tuning_train_dataset":{"value":"../data/processed/sepsis/sepsis_train_feature_list.pkl"},"patience":{"value":20},"shuffle_batch":{"value":true},"train_bert":{"value":true},"use_test_set":{"value":false}},"runqueue_item_id":"UnVuUXVldWVJdGVtOjU4NTEyNzQzMQ==","logs":["expected_improvement: nan","predicted_value: 0.0","predicted_value_std_dev: nan","success_probability: 1.0","warnings: \nSweep has no valid samples of the metric."],"run_storage_id":"UnVuOnYxOmV6OXdoMzk4Ok1lZEJlcnQtRmluZXR1bmluZy1GaXJzdDpwcmFudHp1bGktaGFzc28tcGxhdHRuZXItaW5zdGl0dXQ="}]
 predicted_value: 0.0
 predicted_value_std_dev: nan
 success_probability: 1.0
 warnings:  Sweep has no valid samples of the metric.

 Launched new run ez9wh398 (apricot-sweep-1)
 Sending commands to agent ngtqkr2l: [{"run_id":"xfpd2pg1","program":"","type":"run","args":{"batch_size":{"value":16},"bert_base_path":{"value":"../data/processed/trained_tensorflow_small"},"bert_checkpoint":{"value":4000000},"classifier":{"value":"RNN"},"classifier_dropout_rate":{"value":0.3512012198142431},"classifier_num_hidden_layers":{"value":8},"classifier_size_hidden_layers":{"value":64},"drop_last_batch":{"value":true},"epochs":{"value":18.34083677375235},"lr":{"value":0.00009148373389813548},"min_delta":{"value":0.001},"path_fine_tuning_train_dataset":{"value":"../data/processed/sepsis/sepsis_train_feature_list.pkl"},"patience":{"value":20},"shuffle_batch":{"value":true},"train_bert":{"value":true},"use_test_set":{"value":false}},"runqueue_item_id":"UnVuUXVldWVJdGVtOjU4NTEyODEwMg==","logs":["expected_improvement: nan","predicted_value: -0.000000e+00","predicted_value_std_dev: nan","success_probability: 1.0","warnings: "],"run_storage_id":"UnVuOnYxOnhmcGQycGcxOk1lZEJlcnQtRmluZXR1bmluZy1GaXJzdDpwcmFudHp1bGktaGFzc28tcGxhdHRuZXItaW5zdGl0dXQ="}]
 predicted_value: -0.000000e+00
 predicted_value_std_dev: nan
 success_probability: 1.0
 warnings:

 Launched new run xfpd2pg1 (classic-sweep-2)
 Sending commands to agent ngtqkr2l: [{"run_id":"ksh1qy7k","program":"","type":"run","args":{"batch_size":{"value":64},"bert_base_path":{"value":"../data/processed/trained_tensorflow_small"},"bert_checkpoint":{"value":4000000},"classifier":{"value":"RNN"},"classifier_dropout_rate":{"value":0.16418947328782724},"classifier_num_hidden_layers":{"value":1},"classifier_size_hidden_layers":{"value":32},"drop_last_batch":{"value":true},"epochs":{"value":21.83794195228878},"lr":{"value":0.0000829806094904379},"min_delta":{"value":0.001},"path_fine_tuning_train_dataset":{"value":"../data/processed/sepsis/sepsis_train_feature_list.pkl"},"patience":{"value":20},"shuffle_batch":{"value":true},"train_bert":{"value":true},"use_test_set":{"value":false}},"runqueue_item_id":"UnVuUXVldWVJdGVtOjU4NTEzMjgwOA==","logs":["expected_improvement: 8.936608e-05","predicted_value: 0.000000e+00","predicted_value_std_dev: 2.240075e-10","success_probability: 5.000000e-01","warnings: \n Some dimmensions of kernel are close to their bounds (bad fit), the next sample will be a random sample within parameter space"],"run_storage_id":"UnVuOnYxOmtzaDFxeTdrOk1lZEJlcnQtRmluZXR1bmluZy1GaXJzdDpwcmFudHp1bGktaGFzc28tcGxhdHRuZXItaW5zdGl0dXQ="}]
 predicted_value: 0.000000e+00
 predicted_value_std_dev: 2.240075e-10
 success_probability: 5.000000e-01
 warnings:   Some dimmensions of kernel are close to their bounds (bad fit), the next sample will be a random sample within parameter space
```

### Expected Behavior

The error message is quite undetailed, and besides that, I can't imagine that in this relatively large parameter space, a parameter is already close to its bounds in the second/third iteration (I tested this configuration sometimes, and every time this is what happens after 2-3 runs).

Is there any way to determine which parameter is close to their bounds?

Besides that, shouldn't Bayesian search optimize all parameters even though one parameter is close to its bounds it should keep this parameter and optimize the rest till either all parameters are close to their bounds (the limits of parameters can be an optimum too) or a local (may even global) optimum is found in the given parameter space.

I also couldn't find anything in the documentation about the implementation of Bayesian search and even less about this specific error:

`Some dimmensions of kernel are close to their bounds (bad fit), the next sample will be a random sample within parameter space`

Some more details here would be great so we can fix this (it could also be that the Bayesian search implementation just is buggy?!)

### Steps To Reproduce

1. Use some arbitrary large enough BERT model
2. Define a metric to optimize on (in my case AUROC with values between 0.5 - 1.0)
3. Define a sweep config similar to the one I provided
4. Start the sweep

### Environment

Version:
Python 3.12.4
wandb 0.17.5

**Comment by UliPrantz** [comment]
Hi there

So this is the sweep initialization:
```python
sweep_config = {
    "method": "bayes",
    "run_cap": 200,
    "metric": {
        "name": "validation_auroc_score.max",
        "goal": "maximize"
    },
    "parameters": {
        "epochs": {
            "distribution": "normal",
            "mu": 15,
            "sigma": 5
        },
        "lr": {
            "distribution": "uniform",
            "min": 1e-7,
            "max": 1e-4
        },
        "batch_size": {
            "distribution": "categorical",
            "values": [16, 32, 64, 128]
        },
        "shuffle_batch": {
            "value": True
        },
        "drop_last_batch": {
            "value": True
        },
        "bert_base_path": {
            "value": "/path/to/model/weights"
        },
        "bert_checkpoint": {
            "distribution": "categorical",
            "values": [1_000_000, 2_000_000, 3_000_000, 4_000_000, 5_000_000]
        },
        "train_bert": {
            "value": True
        },
        "classifier": {
            "value": "RNN"
        },
        "classifier_num_hidden_layers": {
            "distribution": "int_uniform",
            "min": 1,
            "max": 10
        },
        "classifier_size_hidden_layers": {
            "distribution": "categorical",
            "values": [16, 32, 64, 128]
        },
        "classifier_dropout_rate": {
            "distribution": "uniform",
            "min": 0.1,
            "max": 0.8
        },
        "patience": {
            "value": 20
        },
        "min_delta": {
            "value": 1e-3
        },
        "path_fine_tuning_train_dataset": {
            "value": "/path/to/training/data"
        },
        "use_test_set": {
            "value": False
        }
    }
}

wandb.login()
sweep_id = wandb.sweep(sweep_config, project="SomeProjectName")
print(f"Running sweep with id: {sweep_id}")
wandb.agent(sweep_id, run_configuration)
```

Also this is how my `run_configuration` function looks like, where also the respective evaluation values are logged:
```python
def run_configuration(config=None, wandb_project_name=None, run_name=None) -> None:    
    # init the wandb_config
    run = wandb.init(
        project=wandb_project_name,
        name=run_name,
        config=config
    )
    config = wandb.config
    
    # get all the needed values from the config in there respective variables (removed this part)

    wandb.define_metric("seconds_per_epoch_and_validation", summary="mean")
    wandb.define_metric("trainings_loss", summary="min")
    wandb.define_metric("validation_loss", summary="min")
    wandb.define_metric("validation_auroc_score", summary="max")
    if use_test_set:
        wandb.define_metric("test_loss", summary="min")
        wandb.define_metric("test_auroc_score", summary="max")
    
    for epoch in range(1, epochs+1):
        
        # train model and evaluate model here on the data provided (removed this part)

        wandb.log(
            {
                "trainings_loss": traininings_loss,
                "validation_loss": validation_loss,
                "validation_auroc_score": validation_sklearn_auroc_score,
                **({"test_loss": test_loss} if use_test_set else {}),
                **({"test_auroc_score": test_sklearn_auroc_score} if use_test_set else {}),
                "seconds_per_epoch_and_validation": elapsed_time
                
            },
            commit=True
        )
```

Here are some links for sweeps where this has happened:

- https://wandb.ai/prantzuli-hasso-plattner-institut/MedBert-Sweep-Run-1/sweeps/kug5i1gd?nw=nwuserprantzuli
- https://wandb.ai/prantzuli-hasso-plattner-institut/MedBert-Sweep-Run-1/sweeps/zfjta821?nw=nwuserprantzuli
- https://wandb.ai/prantzuli-hasso-plattner-institut/MedBert-Sweep-Run-1/sweeps/qpsp07j3?nw=nwuserprantzuli
- https://wandb.ai/prantzuli-hasso-plattner-institut/MedBert-Sweep-Run-1/sweeps/qzukw8gm?nw=nwuserprantzuli
- https://wandb.ai/prantzuli-hasso-plattner-institut/MedBert-Sweep-Run-1/sweeps/66mjkac6?nw=nwuserprantzuli

Im using the WandB hosted version (URL: https://wandb.ai/) therefore I believe the `/system-admin` endpoint is not accessible for me. This is my profile URL though: https://wandb.ai/prantzuli

**Comment by boshwandb** [comment]
Thank you very much for sharing the info. By checking your agent logs for these jobs, I do see this warning appeared multiple times, but fortunately they didn't crash your runs. I agree with you that the warnings are not very detailed and hard to interpret what is going on. I have escalated to our development team on this issue. 
In the mean time, you might find some insights from the [source code](https://github.com/wandb/sweeps/blob/master/src/sweeps/bayes_search.py/) where bayesian search is implemented. The warning is under def next_sample().

**Comment by boshwandb** [comment]
No problem. I’ll then go ahead and close this ticket for now but please feel free to re-open it if you have any other question related.
